src/app/index.ts

```diff
@@ -404,6 +453,90 @@ module.exports = class TheiaExtension extends Base {
             );
         }
 
+        /** DiagramEditor */
+        if (this.params.extensionType === ExtensionType.DiagramEditor) {
+            var done = this.async();
+            download('https://github.com/eclipse-glsp/glsp-examples/archive/master.tar.gz', 'tmp', {
```

Comment:
I'm not really fond of depending on external repos this way. I'm not sure this is appropriate for a `theia` extension generator.

Reply:
I agree this is not ideal. For example we have the `vscode-ripgrep` dependency in the main repo that does this, to fetch a platform-appropriate `ripgrep` binary from GH downloads, and is a rather consistent source of misc problems, like getting the download to work behind some corporate firewalls.

Do we know if the `download` dependency will respect proxy environment settings?

Reply:
Exchanged the `download` dependency and replaced it with the more common `request` and `tar` packages.